internal/providers/terraform/google/dns_record_set.go

```diff
@@ -0,0 +1,45 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetDNSRecordSetRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "google_dns_record_set",
+		RFunc: NewDNSRecordSet,
+	}
+}
+
+func NewDNSRecordSet(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	var monthlyQueries *decimal.Decimal
+
+	if u != nil && u.Get("monthly_queries").Exists() {
+		monthlyQueries = decimalPtr(decimal.NewFromInt(u.Get("monthly_queries").Int()))
+	}
+
+	return &schema.Resource{
+		Name: d.Address,
+		CostComponents: []*schema.CostComponent{
+			{
+				Name:            "Queries",
+				Unit:            "queries",
+				UnitMultiplier:  1000000,
+				MonthlyQuantity: monthlyQueries,
+				ProductFilter: &schema.ProductFilter{
+					VendorName:    strPtr("gcp"),
+					Region:        strPtr("global"),
+					Service:       strPtr("Cloud DNS"),
+					ProductFamily: strPtr("Network"),
+					AttributeFilters: []*schema.AttributeFilter{
+						{Key: "description", Value: strPtr("DNS Query (port 53)")},
+					},
+				},
+				PriceFilter: &schema.PriceFilter{
+					StartUsageAmount: strPtr("0"),
+				},
+			},
+		},
+	}
+}
```

Comment:
Would you not want to have the queries against the overall domain as appose to the individual records. Just thinking about how many records could be created. In this context if you generate a query usage per record set the output could get quite verbose. Domains could have lots of records. Maybe there's no right or wrong way.

Reply:
I initially put the queries cost component in the dns_managed_zone as you mentioned, but @aliscott pointed out that users might have the zone managed outside of terraform (this is why he had split the route_53 zone and query pricing). I'm ok changing this (and the route_53 resource) but leaning towards leaving it as is as so far we've tried to map each TF resource to a set of prices 🤷